### PR TITLE
Fix `gr.Label` styling and a11y markup

### PR DESCRIPTION
**Pull request by whitphx**
## Description

When a label name is long, the text is not aligned to left properly.

<table>
<tr>
<th><code>main</code></th>
<th>This PR</th>
</tr>
<tr>
<td>

</td>
<td>

</td>
</tr>
</table>

The sample code is
```python
import gradio as gr


demo = gr.Interface(
    fn=lambda x: x,
    inputs=gr.Textbox(),
    outputs=gr.Label({
        'tiger, Panthera tigris': 0.632695734500885,
        'tiger cat': 0.3634825646877289,
        'lion, king of beasts, Panthera leo': 0.00045060308184474707
    })
)

demo.launch()
```

During the fix above, I also did some refactoring and fixed some ARIA markups.

### CODE REVIEW

File: js/label/shared/Label.svelte [hunk 1]
```diff
@@ -35,27 +42,35 @@
 			>
 				<div class="inner-wrap">
 					<meter
-						aria-labelledby="meter-text"
+						aria-labelledby={get_aria_referenceable_id(
+							`meter-text-${confidence_set.label}`
+						)}
```
Comment (whitphx) [inline js/label/shared/Label.svelte:47]: Fixed this ARIA error:
CleanShot 2024-04-18 at 12 42 07@2x

File: js/label/shared/Label.svelte [hunk 2]
```diff
@@ -35,27 +42,35 @@
...
-						<dt id={`meter-text-${confidence_set.label}`} class="text">
+						<dt
+							id={get_aria_referenceable_id(
+								`meter-text-${confidence_set.label}`
+							)}
+							class="text"
+						>
 							{confidence_set.label}
 						</dt>
-						{#if value.confidences}
```
Comment (whitphx) [inline js/label/shared/Label.svelte:53]: `if value.confidences` is already check by the outer if-block, so this can be removed?

**Review by abidlabs — APPROVED** [review]
Changes look good, thanks @whitphx! 

My only suggestion would be to add a stories file `js/label/Label.stories.svelte` so that we don't regress these fixes in the future

**Comment by whitphx** [comment]
@abidlabs Thanks! Yes that's important. Will add a story file.

**Comment by abidlabs** [comment]
Thanks for adding @whitphx!

**Pull request merged**